### ISSUE TITLE
Notebook mode filter error using filter by count: Operator does not exist: bigint >= character varying 

### ISSUE THREAD

**Issue by tiwariayush** [metabase/metabase#11223]
**Describe the bug**
When one tries to filter using `Count`, you get to chose a text in the filter and then it fails with an error. Detailed explanation with step by step reproducing below. Also, I have added a hack which I use to overcome the issue in the extra context section. 

**Logs**

```
PSQLException: 
Oct 29 07:27:52 metabase-dev app/web.1:  Message: ERROR: operator does not exist: bigint > character varying 
Oct 29 07:27:52 metabase-dev app/web.1:   Indice : No operator matches the given name and argument type(s). You might need to add explicit type casts. 
Oct 29 07:27:52 metabase-dev app/web.1:   Position : 440 
Oct 29 07:27:52 metabase-dev app/web.1:  SQLState: 42883 
Oct 29 07:27:52 metabase-dev app/web.1:  Error Code: 0 
Oct 29 07:27:52 metabase-dev app/web.1: 10-29 14:27:52 ERROR sql-jdbc.execute :: nil 
Oct 29 07:27:53 metabase-dev app/web.1: 10-29 14:27:52 WARN middleware.process-userland-query :: Erreur de la requête {:status :failed, 
Oct 29 07:27:53 metabase-dev app/web.1:  :class org.postgresql.util.PSQLException, 
Oct 29 07:27:53 metabase-dev app/web.1:  :error "ERROR: operator does not exist: bigint > character varying\n  Indice : No operator matches the given name and argument type(s). You might need to add explicit type casts.\n  Position : 440", 
Oct 29 07:27:53 metabase-dev app/web.1:  :stacktrace 
```

**To Reproduce**
1. You need to use Postgres DB.
2. Go to `Ask a Question` >> `Custom Question`
2. Create a question for a table where you can use group by
2. Add `Count` in summarize and `group by` with a column
3. Scroll down to `Filter` button after the summarize and add a filter on `Count` (in my case, I did `Count is greater than 2`)
4. Press `Visualize` and See the error

**Expected behavior**
Result grouped by whichever column mentioned before filtered by count greater than 2 

**Screenshots**
*Creating summarize `Count` grouped by a column*
![image](https://user-images.githubusercontent.com/5693204/67778005-e52a3b80-fa62-11e9-9d85-6655a34ea186.png)

*You can notice that in filter by `Count`, using `Greater than`, it asks for **Enter some text***
![image](https://user-images.githubusercontent.com/5693204/67778068-f70bde80-fa62-11e9-80dd-7a96b2623781.png)

*The error being displayed*
![image](https://user-images.githubusercontent.com/5693204/67778301-46eaa580-fa63-11e9-9460-ea6802790aa9.png)

**Information about your Metabase Installation:**

- Your browser and the version: Firefox 70.0 (64-bit)
- Your operating system: Ubuntu 18.04
- Your databases: Postgres
- Metabase version: 0.33.4
- Metabase hosting environment: Heroku
- Metabase internal database: H2 (default)

**Severity**
It is quite blocking to create custom questions.

**Additional context**
*If you look create a custom column for count and then use this custom colum to do the filtering, it works perfectly*
![image](https://user-images.githubusercontent.com/5693204/67778446-7d282500-fa63-11e9-9ceb-daa1bfe9f413.png)

*In this case using a custom `Count` column works fine as it shows `Enter a number`*
![image](https://user-images.githubusercontent.com/5693204/67778503-929d4f00-fa63-11e9-8ba7-172854b165dd.png)

Seems like there has been discussion (https://github.com/metabase/metabase/issues/5632) reagrding this in past and it should be fixed by https://github.com/metabase/metabase/pull/10809
but it still lingers.

**Comment by alichaudry** [comment]
Similar issue for me. On doing a filter on a count generated by Metabase, I'm getting:

`ERROR: operator does not exist: bigint > character varying Hint: No operator matches the given name and argument type(s). You might need to add explicit type casts. Position: 538`

In the query editor it looks like the SQL that Metabase is generating is creating the filter against a string instead of an int: 

```
WHERE "source"."count" > '9' LIMIT 1048576
```

**Comment by flamber** [comment]
This was fixed in 0.34.3
